src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -584,7 +583,8 @@ uct_dc_mlx5_iface_dci_put(uct_dc_mlx5_iface_t *iface, uint8_t dci_index)
 
     ucs_assert(dci_index != UCT_DC_MLX5_EP_NO_DCI);
 
-    if (uct_dc_mlx5_iface_is_policy_shared(iface)) {
+    if (uct_dc_mlx5_iface_is_policy_shared(iface) ||
+        uct_dc_mlx5_is_hw_dci(iface, dci_index)) {
```

Comment:
can we use is_dci_shared?

Reply:
We can't because `is_dci_shared` is not checking wether the dci is NULL (it will fail in random policy for example)

Reply:
how can the DCI be NULL, we are releasing it after it was used?